src/service/viewport-impl.js

```diff
@@ -1331,28 +1341,28 @@ export function updateViewportMetaString(currentValue, updateParams) {
 
 
 /**
- * @param {!Window} window
+ * @param {!./ampdoc-impl.AmpDoc} ampdoc
  * @return {!Viewport}
  * @private
  */
-function createViewport_(window) {
-  const viewer = installViewerService(window);
+function createViewport(ampdoc) {
+  const viewer = installViewerService(ampdoc.win);
   let binding;
-  if (viewer.getViewportType() == 'natural-ios-embed') {
-    binding = new ViewportBindingNaturalIosEmbed_(window);
+  if (ampdoc.isSingleDoc() &&
+          viewer.getViewportType() == 'natural-ios-embed') {
+    binding = new ViewportBindingNaturalIosEmbed_(ampdoc.win);
   } else {
-    binding = new ViewportBindingNatural_(window, viewer);
+    binding = new ViewportBindingNatural_(ampdoc.win, viewer);
   }
-  return new Viewport(window, binding, viewer);
+  return new Viewport(ampdoc, binding, viewer);
 }
 
 
 /**
- * @param {!Window} window
+ * @param {!./ampdoc-impl.AmpDoc} ampdoc
  * @return {!Viewport}
  */
-export function installViewportService(window) {
-  return /** @type !Viewport} */ (getService(window, 'viewport', () => {
-    return createViewport_(window);
-  }));
+export function installViewportServiceForDoc(ampdoc) {
+  return /** @type !Viewport} */ (getServiceForDoc(ampdoc, 'viewport',
```

Comment:
I fixed typedef itself. And as mentioned, the need for it at all is being removed right now by @erwinmombay